packages/in_app_purchase/in_app_purchase_ios/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart

```diff
@@ -70,18 +74,39 @@ class SKPaymentQueueWrapper {
   ///
   /// Call this method when the first listener is subscribed to the
   /// [InAppPurchaseIosPlatform.purchaseStream].
-  Future startObservingTransactionQueue() async =>
-      await channel.invokeListMethod<void>(
-          '-[SKPaymentQueue startObservingTransactionQueue]');
+  Future startObservingTransactionQueue() => channel
+      .invokeMethod<void>('-[SKPaymentQueue startObservingTransactionQueue]');
 
   /// Instructs the iOS implementation to remove the transaction observer and
   /// stop listening to it.
   ///
   /// Call this when there are no longer any listeners subscribed to the
   /// [InAppPurchaseIosPlatform.purchaseStream].
-  Future stopObservingTransactionQueue() async =>
-      await channel.invokeListMethod<void>(
-          '-[SKPaymentQueue stopObservingTransactionQueue]');
+  Future stopObservingTransactionQueue() => channel
+      .invokeMethod<void>('-[SKPaymentQueue stopObservingTransactionQueue]');
+
+  /// Sets an implementation of the [SKPaymentQueueDelegateWrapper].
+  ///
+  /// The [SKPaymentQueueDelegateWrapper] can be used to inform iOS how to
+  /// finish transactions when the storefront changes or if the price consent
+  /// sheet should be displayed when the price of a subscription has changed. If
+  /// no delegate is registered iOS will fallback to it's default configuration.
+  /// See the documentation on StoreKite's [`-[SKPaymentQueue delegate:]`](https://developer.apple.com/documentation/storekit/skpaymentqueue/3182429-delegate?language=objc).
+  ///
+  /// When set to `null` the payment queue delegate will be removed and the
+  /// default behaviour will apply (see [documentation](https://developer.apple.com/documentation/storekit/skpaymentqueue/3182429-delegate?language=objc)).
+  Future setDelegate(SKPaymentQueueDelegateWrapper? delegate) async {
+    if (delegate == null) {
+      await channel.invokeMethod<void>('-[SKPaymentQueue removeDelegate]');
+      paymentQueueDelegateChannel.setMethodCallHandler(null);
+    } else {
+      await channel.invokeMethod<void>('-[SKPaymentQueue registerDelegate]');
+      paymentQueueDelegateChannel
+          .setMethodCallHandler(handlePaymentQueueDelegateCallbacks);
+    }
+
+    _paymentQueueDelegate = delegate;
+  }
```

Comment:
What if you register 2 delegates?
Shouldn't this contain a check `if(_paymentQueueDelegate != null)` and throw an exception "Already a delegate registered".

Reply:
If you register two delegates the existing one is replaced with the new one. There can always only be one delegate active. 

If you would add the check you would never be able to replace it if you wanted (although it is not likely that this is necessary).